src/components/MessageList/utils/getGroupStyles.js

```diff
@@ -0,0 +1,81 @@
+export const getGroupStyles = (messagesWithDates, noGroupByUser) => {
+  const numberOfMessages = messagesWithDates.length;
+  const messageGroupStyles = {};
+
+  const messages = [...messagesWithDates];
+
+  for (let i = 0; i < numberOfMessages; i++) {
```

Comment:
might be helpful to put a comment on this for loop describing what's happening for future reference

Reply:
Done, was shifted over from the component but I added some context with comments.